src/Microsoft.AspNetCore.SignalR.Protocols.Json/Protocol/JsonHubProtocol.cs

```diff
@@ -225,6 +226,14 @@ private HubMessage ParseMessage(Utf8BufferTextReader textReader, IInvocationBind
                                             catch (Exception ex)
                                             {
                                                 argumentBindingException = ExceptionDispatchInfo.Capture(ex);
+
+                                                // Could be at any point in argument array JSON when an error is thrown
+                                                // Read until the end of the argument JSON array
+                                                while (reader.Depth == initialDepth && reader.TokenType == JsonToken.StartArray ||
```

Comment:
This matters because we catch the error right? I guess today the stream is corrupted at this point since we don't consume everything? 

I'm trying to figure out why this is the only place we need to do this.